src/main/java/hudson/plugins/git/util/GitUtils.java

```diff
@@ -190,7 +219,11 @@ public Revision sortBranchesForRevision(Revision revision, List<BranchSpec> bran
                         }
 
                     } finally {
-                        walk.release();
+                        try {
+                            walk.release();
```

Comment:
So basically, yeah, I think at least in some JVMs after tiered compilation has kicked in.

Basically if you know the method is never going to be there on that JVM, you want to let the JVM optimise out trying.

Something like:

```
public static void close(walker) {
  try {
    _release(walker);
  } catch (LinkageError e) {
    reflectionClose(walker);
  }
}

/* hotspot curses upon you */
private static void _release(walker) {
  walker.release()
}
```

Should do it.